### PR TITLE
Stored Favorite / Membership Counts

### PR DESCRIPTION
**Pull request by shiva-menta**
PR to cache favorite / membership counts for clubs. The alternative is recalculating this for every query.

### CODE REVIEW

File: backend/clubs/management/commands/update_club_counts.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                temp_favorite_count=Count("favorite", distinct=True),
+                temp_membership_count=Count(
+                    "membership", distinct=True, filter=Q(active=True)
+                ),
+            )
+
+            for club in queryset:
+                club.favorite_count = club.temp_favorite_count
+                club.membership_count = club.temp_membership_count
+                club.save()
```
Comment (rm03) [inline backend/clubs/management/commands/update_club_counts.py:22]: consider using `bulk_update()`

**Comment by rohangpta** [comment]
Could be an interesting exercise to upgrade to 5.0 and also update related deps (which will include upgrading the Python version which has a host of its own benefits), so that we can take advantage of [this](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.GeneratedField). Wdyt?

File: backend/clubs/management/commands/update_club_counts.py (hunk 1 above)
Comment (shiva-menta) [reply backend/clubs/management/commands/update_club_counts.py:22]: Yup good call.

File: backend/pennclubs/settings/base.py [hunk 2]
```diff
@@ -18,6 +18,9 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
+# Django Setting Environment Variable
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "pennclubs.settings.base")
```
[the remark this answers is not in the document]
Comment (shiva-menta) [reply backend/pennclubs/settings/base.py:22]: Not entirely sure honestly I think I may have added it on accident. Removing.

[the remark this answers is not in the document]
Comment (shiva-menta) [reply backend/pennclubs/settings/base.py:22]: Oh never mind I needed that to run the test cases for some reason, won't include it in PR though.

**Pull request merged**